etcd/src/test/scala/io/buoyant/etcd/KeyTest.scala

```diff
@@ -0,0 +1,441 @@
+            rsp.content = writeJson(e)
+            rsp
+        }
+      } else {
+        rsp.status = Status.InternalServerError
+        info(s"method=${req.method} path=$path params=$params")
+        Future.value(rsp)
+      }
+    }
+
+  private[this] def mkEtcd(
+    filter: Filter[Request, Response, Request, Response] = Filter.identity
+  )(handle: PartialFunction[(Method, Path, Params), Any]) =
+    new Etcd(filter andThen mkClient(handle))
+
+  private[this] def mkKey(
+    key: Path,
+    filter: Filter[Request, Response, Request, Response] = Filter.identity
+  )(handle: PartialFunction[(Method, Params), Any]): Key = {
+    val uri = Path.Utf8("v2", "keys") ++ key
+    val etcd = mkEtcd(filter) {
+      case (method, `uri`, params) if handle.isDefinedAt(method, params) =>
+        handle((method, params))
+    }
+    etcd.key(key)
+  }
+
+  private[this] def serve(method: Method, rsp: Response, v: Any): Future[Response] =
+    v match {
+      case f: Future[_] =>
+        f.flatMap(serve(method, rsp, _))
+
+      case Some(v) =>
+        serve(method, rsp, v)
+
+      case op: NodeOp =>
+        rsp.content = writeJson(NodeOp.Json(op))
+        addState(op.etcd, rsp)
+        rsp.status = (rsp.status, method) match {
+          case (Status.Ok, Method.Put | Method.Post) => Status.Created
+          case (status, _) => status
+        }
+        Future.value(rsp)
+
+      case v: Version =>
+        rsp.content = writeJson(v)
+        Future.value(rsp)
+
+      case idk =>
+        rsp.status = Status.InternalServerError
+        rsp.contentString = idk.toString
+        Future.value(rsp)
+    }
+
+  private[this] def addState(etcd: Etcd.State, msg: Message): Unit = {
+    msg.headerMap("X-Etcd-Index") = etcd.index.toString
+    msg.headerMap("X-Etcd-Cluster-Id") = etcd.clusterId
+  }
+
+  private[this] object Recursive {
+    def unapply(params: Params): Option[Boolean] =
+      Some(getParam(params, "recursive").exists(_ == "true"))
+  }
+  private[this] object DirParam {
+    def unapply(params: Params): Option[Boolean] =
+      Some(getParam(params, "dir").exists(_ == "true"))
+  }
+  private[this] object DirRecursive {
+    def unapply(params: Params): Option[(Boolean, Boolean)] = {
+      val DirParam(dir) = params
+      val Recursive(recursive) = params
+      Some((dir, recursive))
+    }
+  }
+
+  private[this] object Watch {
+    def unapply(params: Params): Option[(Boolean, Option[Int])] = {
+      val Recursive(recursive) = params
+      val idx = getParam(params, "wait").find(_ == "true") flatMap { _ =>
+        getParam(params, "waitIndex") flatMap { waitIndex =>
+          Try(waitIndex.toInt).toOption
+        }
+      }
+      Some((recursive, idx))
+    }
+  }
+
+  test("Etcd.version") {
+    val uri = Path.Utf8("version")
+    val version = Version("a.b.c.d", "a")
+    val etcd = mkEtcd() { case (Method.Get, `uri`, _) => version }
+    val v = etcd.version()
+    assert(Await.result(v, 250.millis) == version)
```

Comment:
Would it be worth mixing Awaits into this test class? 250ms is the default, so you could just write this as `assert(await(v) == version)`.